extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -842,22 +826,9 @@ private static void throwIfQueryIsNotSuccessful(String queryId, TaskStatusPlus s
 
   private void contextChecks(QueryContext queryContext)
   {
-    ExecutionMode executionMode = queryContext.getEnum(
-        QueryContexts.CTX_EXECUTION_MODE,
-        ExecutionMode.class,
-        null
-    );
+    ExecutionMode executionMode = queryContext.getEnum(QueryContexts.CTX_EXECUTION_MODE, ExecutionMode.class, null);
     if (ExecutionMode.ASYNC != executionMode) {
-      throw DruidException.forPersona(DruidException.Persona.USER)
-                          .ofCategory(DruidException.Category.INVALID_INPUT)
-                          .build(
-                              StringUtils.format(
-                                  "The statement sql api only supports sync mode[%s]. Please set context parameter [%s=%s] in the context payload",
-                                  ExecutionMode.ASYNC,
-                                  QueryContexts.CTX_EXECUTION_MODE,
-                                  ExecutionMode.ASYNC
-                              )
-                          );
+      throw DruidException.defensive("[%s] is not supported. It should not be set", QueryContexts.CTX_EXECUTION_MODE);
```

Comment:
```suggestion
      throw InvalidInput.exception(
                                  "The statement sql api only supports execution mode[%s]. Please set context parameter [%s=%s] in the context payload",
                                  ExecutionMode.ASYNC,
                                  QueryContexts.CTX_EXECUTION_MODE,
                                  ExecutionMode.ASYNC
                              );
```

This should be invalid input exception cause we need to tell the user to set another execution mode.